### DISCUSSION TITLE
Context menu

### DISCUSSION THREAD

**Discussion by arunnbabu81** [buggyj/tiddlyclip#74] [Ideas]
It would be better if the context menu could be modified like this  - nested or cascading context menu ( don't know whether that's the right term or not)

![image](https://user-images.githubusercontent.com/67494083/158671847-0a90e91b-7e65-46be-b76b-f0df85733f04.png)
Is that a feasible option?

I have a huge list of clip rules, so such a nested context menu will be very useful

**Comment by buggyj** [comment]
I find it difficult to select this type of cascading menu items, that is why the TiddlyClipConfig is split into a number of tables instead - it allows the rules to be organized.

**Reply by arunnbabu81** [reply]
> I find it difficult to select this type of cascading menu items, that is why the TiddlyClipConfig is split into a number of tables instead - it allows the rules to be organized.

Can it be added as a configurable option if not default

**Reply by buggyj** [reply]
I will be rewritten the menu sometime this year. I will look to include this then.

**Reply by buggyj** [reply]
There is a pre-release of the new version for chrome here https://github.com/buggyj/tiddlyclip3 - it has cascading menus.

**Reply by arunnbabu81** [reply]
> There is a pre-release of the new version for chrome here https://github.com/buggyj/tiddlyclip3 - it has cascading menus.

Thank you @buggyj I will check it and give feedback. I was able to install the extension in chrome. But I forgot how to do it in firefox. Is there any instructions anywhere in the site for installation in firefox.

**Reply by buggyj** [reply]
It is only for chrome at the moment - firefox has yet to fully implement V3

**Reply by arunnbabu81** [reply]
Is clipping from pdf embedded in tiddlers supported in chrome ? Its not working when I tested with the new extension.

**Reply by buggyj** [reply]
I imagine that it is not supported as discussed here:
https://github.com/buggyj/tiddlyclip-plugin/discussions/107

**Reply by arunnbabu81** [reply]
> I imagine that it is not supported as discussed here: [buggyj/tiddlyclip-plugin#107](https://github.com/buggyj/tiddlyclip-plugin/discussions/107)

I can clip from pdf tiddlers in Firefox. Only in chrome it's not working and it was like that previously also I guess

**Reply by buggyj** [reply]
you could copy the text (control-c) and this would appear in the variable `@clip`

**Reply by arunnbabu81** [reply]
Thanks for the suggestion. It works.
